specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualMachines.json

```diff
@@ -0,0 +1,801 @@
+                  },
+                  "description": "SSH - contains settings related to ssh configuration"
+                }
+              },
+              "description": "LinuxConfiguration - linux specific configuration values for the virtual machine"
+            },
+            "osType": {
+              "type": "string",
+              "description": "OsType - string specifying whether the OS is Linux or Windows"
+            },
+            "windowsConfiguration": {
+              "type": "object",
+              "properties": {
+                "enableAutomaticUpdates": {
+                  "type": "boolean",
+                  "description": "Whether to EnableAutomaticUpdates on the machine"
+                },
+                "ssh": {
+                  "type": "object",
+                  "properties": {
+                    "publicKeys": {
+                      "type": "array",
+                      "items": {
+                        "type": "object",
+                        "properties": {
+                          "keyData": {
+                            "type": "string",
+                            "x-ms-secret": true,
+                            "description": "KeyData - SSH public key certificate used to authenticate with the VM through ssh. The key needs to be at least 2048-bit and in ssh-rsa format. <br><br> For creating ssh keys, see [Create SSH keys on Linux and Mac for Li      nux VMs in Azure](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-linux-mac-create-ssh-keys?toc=%2fazure%2fvirtual-machines%2flinux%2ftoc.json)."
+                          },
+                          "path": {
+                            "type": "string",
+                            "description": "Path - Specifies the full path on the created VM where ssh public key is stored. If the file already exists, the specified key is appended to the file. Example: /home/user/.ssh/authorized_keys"
+                          }
+                        }
+                      },
+                      "description": "PublicKeys - The list of SSH public keys used to authenticate with linux based VMs.",
+                      "x-ms-identifiers": []
+                    }
+                  },
+                  "description": "SSH Configuration"
+                },
+                "timeZone": {
+                  "type": "string",
+                  "description": "TimeZone for the virtual machine"
+                }
+              },
+              "description": "Windows Configuration for the virtual machine "
+            }
+          },
+          "description": "OsProfile - describes the configuration of the operating system and sets login data"
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "name of the object to be used in moc"
+        },
+        "securityProfile": {
+          "type": "object",
+          "properties": {
+            "enableTPM": {
```

Comment:
>enableTPM [](http://example.com/codeflow?start=13&length=9)

String enum: TPM: { Enabled, Disabled }.

Reply:
Added this workitem to this pending task: https://dev.azure.com/msazure/msk8s/_boards/board/t/ARC-HCI/Stories/?workitem=12679581